### PR TITLE
Closes #30, Closes #17, Closes #27-Initial Repositories and SampleData

### LINKED ISSUE #17: Create model for Path (Creek and Trail)
Create a model for `Path` and its two subclasses, `Creek` and `Trail`.  Will encapsulate the set of geocoordinates required to display a path on the map, with the subclasses providing the necessary styling and tags.  Will contain a function to generate a google `Polyline` object compatible with the Maps api.
DoD:  `Path`, `Creek`, and `Trail` created and able to produce a `Polyline` from sample data.

### LINKED ISSUE #27: Create a Repository for Plant
Create `PlantRepository` to provide application with access to existing instances of `Plant`

### LINKED ISSUE #30: Separate sample data from into new class
Create `SampleData` class to provide selectable static data to repository classes

### PR DESCRIPTION
**Pull request by gomezfox**
The initial model for paths and its repository are complete.  Hardcoded sample data in the repositories has been moved to a separate class, which populates the repos with data.

### CODE REVIEW

**Review by Grant-Buchanan — APPROVED** [review]

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/GeoCoordinate.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package edu.cascadia.mobas.photopoints.model;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.maps.model.LatLng;
+
+
+
+public class GeoCoordinate {
```
Comment (silvioguiso) [inline PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/GeoCoordinate.java:8]: Java has a built-in class, Point, that looks a lot like a simple form of this. Could have potentially used that as a parent. No show-stopper though.

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+
+    // Returns a reference of the Geocoordinates List
+    public List<GeoCoordinate> getGeoCoordinates() {
+        return mGeo;
+    }
+
+
+    // Return integer color value for path display on map
+    private int getLineColor(Context context) {
```
Comment (silvioguiso) [inline PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java:81]: Should we consider doing this with a helper class? I don't know if I like the idea of passing Android components into our model entities.

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                pattern.add(new Dash(5));
+                pattern.add(new Gap(3));
+                return pattern;
+            default: return null;  // default is solid line;
+        }
+    }
+
+
+    // returns path as Google Maps polyline
+    public PolylineOptions getPolylineOptions (Context context) {
```
Comment (silvioguiso) [inline PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java:107]: Same comment as above.

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/GeoCoordinate.java (hunk 1 above)
Comment (gomezfox) [reply PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/GeoCoordinate.java:8]: I attempted to do just that, but it is a final class, can not be used as a parent.

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/ui/map/MapFragment.java [hunk 4]
```diff
@@ -208,3 +200,34 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
         }
     }
 }
+
```
Comment (silvioguiso) [inline PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/ui/map/MapFragment.java:203]: We should not comment old code. Instead, we should remove it. Github is our "original code" bucket :) !

**Review by silvioguiso — APPROVED** [review]

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java (hunk 2 above)
Comment (gomezfox) [reply PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java:81]: Addressed in #52

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java (hunk 3 above)
Comment (gomezfox) [reply PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/model/Path.java:107]: Addressed in #52

File: PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/ui/map/MapFragment.java (hunk 4 above)
Comment (gomezfox) [reply PhotoPoints/app/src/main/java/edu/cascadia/mobas/photopoints/ui/map/MapFragment.java:203]: Removed in #45

**Pull request merged**